File: SwrveSDKCommon/src/main/java/com/swrve/sdk/SwrveFlavour.java
package com.swrve.sdk;

public enum SwrveFlavour {
    AMAZON, CORE, FIREBASE, HUAWEI;

    @Override
    public String toString() {
        switch (this) {
            case AMAZON:
                return "amazon";
            case CORE:
                return "core";
            case FIREBASE:
                return "firebase";
            case HUAWEI:
                return "huawei";
            default:
                throw new IllegalArgumentException();
        }
    }
}

File: SwrveSDKCommon/src/main/java/com/swrve/sdk/SwrveFlavour.kt
package com.swrve.sdk

enum class SwrveFlavour {
    CORE, FIREBASE, HUAWEI;

    override fun toString(): String {
        return when (this) {
            CORE -> "core"
            FIREBASE -> "firebase"
            HUAWEI -> "huawei"
        }
    }
}
